assembly/src/lib.rs

```diff
@@ -104,6 +135,18 @@ impl Assembler {
         &self.kernel
     }
 
+    // STATE MUTATORS
+    // --------------------------------------------------------------------------------------------
+
+    /// Add an instance of the provider to the internal list
+    pub fn replace_module_provider<P>(&mut self, provider: P)
+    where
+        // The static lifetime will bind the type definition, and not instances of it
+        P: ModuleProvider + 'static,
+    {
+        self.module_provider = Box::new(provider);
+    }
```

Comment:
We probably don't need this any more since we can set the `ModuleProvider` at construction time.

Reply:
Agreed. Done!